File: pueblo-golf-tournament-api/Dto/RegistrationDto.cs
using pueblo_golf_tournament_api.Base;
using pueblo_golf_tournament_api.Entities;
using pueblo_golf_tournament_api.Utilities.Enums;

namespace pueblo_golf_tournament_api.Dto
{
    public class RegistrationDto : BaseDto
    {
        public long? TeamId { get; set; }
        public long TeamCaptainId { get; set; }
        public long? TournamentId { get; set; }
        public long? DivisionId { get; set; }
        public RegistrationStatusEnum Status { get; set; }
        public DateTime RegistrationDate { get; set; }
        public long? PaymentId { get; set; }
        public virtual Payment? Payment { get; set; }
    }
    public class RegistrationRawDto
    {
        public long? TeamId { get; set; }
        public long TeamCaptainId { get; set; }
        public long? TournamentId { get; set; }
        public long? DivisionId { get; set; }
        public RegistrationStatusEnum Status { get; set; }
        public DateTime RegistrationDate { get; set; }
    }
}

File: pueblo_golf_tournament_mobile/lib/api/dtos/registration-dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/utilities/enums.dart';

import '../../models/base-model.dart';
part 'registration-dto.g.dart';

@JsonSerializable()
class RegistrationDto extends BaseModel {
  final int teamId;
  final int teamCaptainId;
  final int tournamentId;
  final int divisionId;
  final int status;
  final DateTime registrationDate;
  final int? paymentId;

  factory RegistrationDto.fromJson(Map<String, dynamic> json) =>
      _$RegistrationDtoFromJson(json);

  RegistrationDto(this.teamId, this.teamCaptainId, this.tournamentId,
      this.divisionId, this.status, this.registrationDate, this.paymentId,
      {required super.id});

  Map<String, dynamic> toJson() => _$RegistrationDtoToJson(this);
}
